File: workspaces/clients/php/client/src/Generics/Puth/Context.php
<?php

namespace Puth\Generics\Puth;

use Puth\GenericObject;

class Context extends GenericObject
{
    /**
     * @debug-ts-return-types Browser
     */
    public function createBrowserShim(mixed $page, string $baseUrl): Browser
    {
        return $this->callMethod('createBrowserShim', [$page, $baseUrl]);
    }

    /**
     * @debug-ts-return-types void
     */
    public function destroy(mixed $options): void
    {
        $this->callMethod('destroy', [$options]);
    }

    /**
     * @debug-ts-return-types void
     */
    public function testFailed(): void
    {
        $this->callMethod('testFailed');
    }

    /**
     * @debug-ts-return-types void
     */
    public function saveTemporaryFile(mixed $name, mixed $content): void
    {
        $this->callMethod('saveTemporaryFile', [$name, $content]);
    }
}

File: workspaces/clients/java/client/src/main/java/io/puth/client/remote/Context.java
package io.puth.client.remote;

import io.puth.client.RemoteObject;
import java.util.Map;

/**
* @codegen
*/
public class Context extends RemoteObject {
    public Context(String id, String type, String represents, RemoteObject parent, io.puth.client.Context context) {
        super(id, type, represents, parent, context);
    }

    public Browser createBrowserShim() {
        return this.createBrowserShim(Map.of(), Map.of());
    }

    public Browser createBrowserShim(Map<String, Object> options) {
        return this.createBrowserShim(options, Map.of());
    }

    public Browser createBrowserShim(Map<String, Object> options, Map<String, Object> shimOptions) {
        return (Browser) this.callFunc("createBrowserShim", new Object[]{options, shimOptions});
    }

    public boolean destroy() {
        return this.destroy(Map.of());
    }

    public boolean destroy(Object options) {
        return (boolean) this.callFunc("destroy", new Object[]{options});
    }

    public Object[] getSnapshotsByType(Object type) {
        return (Object[]) this.callFunc("getSnapshotsByType", new Object[]{type});
    }

    public void testFailed() {
        this.callFunc("testFailed", new Object[]{});
    }

    public void testSuccess() {
        this.callFunc("testSuccess", new Object[]{});
    }

    public void testSucceeded() {
        this.callFunc("testSucceeded", new Object[]{});
    }

    public String saveTemporaryFile(Object name, Object content) {
        return (String) this.callFunc("saveTemporaryFile", new Object[]{name, content});
    }

}
